datastream-server/src/test/java/com/linkedin/datastream/server/zk/TestZkAdapter.java

```diff
@@ -758,32 +765,37 @@ public void testDeleteTasksWithPrefix() {
     Mockito.verify(zkClient, Mockito.never()).getChildren(any(), anyBoolean());
 
     List<String> leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(false);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(true);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     updateInstanceAssignment(adapter, adapter.getInstanceName(), Collections.emptyList());
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(false);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(true);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 0);
+    Assert.assertEquals(leftOverTasks.size(), 1);
+
+    // lock root node does not get deleted, once created.
+    adapter.releaseTask(lockTask);
+    leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
+    Assert.assertEquals(leftOverTasks.size(), 1);
```

Comment:
This part seems a little extraneous. 

Should this test really assert that `lock` remains after `releaseTask()`? All it cares about is that `lock` survives `cleanUpOrphanConnectorTasks()`. On the other hand, if we change `ZkAdapter` in the future to have it delete `lock` if it's no longer needed, why should this test start failing then?

Reply:
I feel this part is okay to have. If the behavior changes in the future, whoever is making the change will get alerted and get aware of this existing behavior ( which is not very obvious) and help in either fixing the test or their new behavior.